packages/kbn-dev-utils/src/plugin_list/extract_asciidoc_info.test.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { extractAsciidocInfo } from './extract_asciidoc_info';
+
+it('Returns the info and anchor when there is only one paragraph', () => {
+  const { firstParagraph, anchor } = extractAsciidocInfo(
+    `[[this-is-the-anchor]]
+== I'm the heading!
+
+Hello
+
+I'm an intro paragraph!`
+  );
+
+  expect(firstParagraph).toEqual(`Hello\n\nI'm an intro paragraph!`);
+  expect(anchor).toEqual('this-is-the-anchor');
+});
+
+it('Returns the info and anchor when there are multiple paragraphs without an anchor', () => {
+  const { firstParagraph, anchor } = extractAsciidocInfo(
+    `[[this-is-the-anchor]]
+== Heading here
+
+Intro.
+
+=== Another heading
+
+More details`
+  );
+
+  expect(firstParagraph).toEqual(`Intro.`);
+  expect(anchor).toEqual('this-is-the-anchor');
```

Comment:
Little suggestion, I'm sure you've got your reasons for not preferring this, but wanted to suggest considering inline snapshots

```ts
import dedent from 'dedent';

//...

expect(
  extractAsciidocInfo(dedent`
    [[this-is-the-anchor]]
    == Heading here

    Intro.

    === Another heading

    More details
  `)
).toMatchInlineSnapshot(...autogenerated by jest...)
```